src/engines/nwn2/script/functions_itemproperty.cpp

```diff
@@ -102,6 +105,704 @@ void Functions::getIsItemPropertyValid(Aurora::NWScript::FunctionContext &ctx) {
 		ctx.getReturn() = prop->getIsItemPropertyValid();
 }
 
+void Functions::itemPropertyAbilityBonus(Aurora::NWScript::FunctionContext &ctx) {
+	ItemPropertyType type = kItemPropertyAbilityBonus;
+	uint16 ability = (uint16) ctx.getParams()[0].getInt();
+	uint8 bonus = (uint8) ctx.getParams()[1].getInt();
+
+	// Check the range
+	if (ability >= (uint16) kAbilityMAX || bonus < 1 || bonus > 12)
```

Comment:
We're probably going to need these different item property ranges in other places too, right? Might be useful to create `static constexpr` min/max values for that in src/engines/nwn2/types.h (like the `static const uint32 kClassInvalid` there; that one should be a constexpr now as well).